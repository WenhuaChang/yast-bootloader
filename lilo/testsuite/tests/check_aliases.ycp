{
    include "testsuite.ycp";
    include "lilo/routines.ycp";

    // this one should make no problem
    map liloconf = $[
	        "image":  [
                        $["label" : "image1"],
                        $["label" : "image2", "alias": "image alias"]
                ],
		"other":  [
                        $["label" : "other1", "alias": "other alias"],
                        $["label" : "other2"]
                ]
    ];

    TEST(``(check_aliases(liloconf, "image1", "new_lab", "new_image")), [], nil);

    // complain for duplicitous sec label
    TEST(``(check_aliases(liloconf, "image1", "image2", "")), [], nil);
    TEST(``(check_aliases(liloconf, "image1", "other2", "")), [], nil);

    // no prob
    TEST(``(check_aliases(liloconf, "image2", "image4", "image alias")), [], nil);

    // prob
    TEST(``(check_aliases(liloconf, "image1", "brekeke", "other alias")), [], nil);
    TEST(``(check_aliases(liloconf, "image1", "brekeke", "image alias")), [], nil);

    // alias/label conflict
    TEST(``(check_aliases(liloconf, "image1", "brekeke", "image2")), [], nil);
    TEST(``(check_aliases(liloconf, "image1", "brekeke", "other2")), [], nil);

    TEST(``(check_aliases(liloconf, "other2", "brekeke", "image2")), [], nil);
    TEST(``(check_aliases(liloconf, "other2", "image alias", "adf")), [], nil);


    // alias conflict
    TEST(``(check_aliases(liloconf, "other2", "brekeke", "image alias")), [], nil);
    TEST(``(check_aliases(liloconf, "image2", "ialias", "other alias")), [], nil);

}
