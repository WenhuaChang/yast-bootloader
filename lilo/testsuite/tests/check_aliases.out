Return	
Return	Section label 'image2' is already used.
Return	Section label 'other2' is already used.
Return	
Return	Alias 'other alias' is already used as an alias for section 'other1'.
Return	Alias 'image alias' is already used as an alias for section 'image2'.
Return	Alias 'image2' conflicts with the name of another section.
Return	Alias 'other2' conflicts with the name of another section.
Return	Alias 'image2' conflicts with the name of another section.
Return	Label 'image alias' is already used as an alias for section 'image2'.
Return	Alias 'image alias' is already used as an alias for section 'image2'.
Return	Alias 'other alias' is already used as an alias for section 'other1'.
